tests/csapi/device_lists_test.go

```diff
@@ -160,7 +171,18 @@ func TestDeviceListUpdates(t *testing.T) {
 		barrier := makeBarrier(t, deployment, alice, otherHSName)
 		checkBobKeys := uploadNewKeys(t, bob)
 
-		roomID := alice.MustCreateRoom(t, map[string]interface{}{"preset": "public_chat"})
+		roomID := alice.MustCreateRoom(t, map[string]interface{}{
+			"preset": "public_chat",
+			"initial_state": []map[string]interface{}{
+				{
+					"type":      "m.room.encryption",
+					"state_key": "",
+					"content": map[string]interface{}{
+						"algorithm": "m.megolm.v1.aes-sha2",
+					},
+				},
+			},
+		})
```

Comment:
> Needed for homeservers that only apply E2EE logic (device changes, etc) to shared encrypted rooms, vs. all shared rooms.

So I don't have to guess or make assumptions from the code, can you be more specific with what that means? How does this work for Synapse?

Making this change, makes it work with [babbleserv](/beeper/babbleserv) because xxx?

Does the Matrix spec say anything around this? Does it need a clarification?